elastic/kibana
# [Fleet] Fix config migration from ingestManager to support both xpack.ingestManager and xpack.fleet

## Description 

Fix #111278

Allow the config to specify both `xpack.ingestManager` and `xpack.fleet`. 
Added some test that we were missing here.

I added 7.14.2 as a target version let me know what do you think?

## Release note

Fix a bug when both legacy `xpack.ingestManager` and `xpack.fleet` settings were set `xpack.fleet` was not used.

## How to test

* Set `xpack.ingestManager.registryUrl`
* Set `xpack.fleet.registryProxyUrl`
* Observe that the `registryProxyUrl` setting received by the plugin is set and `registryUrl` is defined too.
